### PR TITLE
Allow injected properties in BaseSpeed selector

### PR DESCRIPTION
**Pull request by simonward**

### CODE REVIEW

File: src/module/rules/rule-element/base-speed.ts [hunk 1]
```diff
@@ -40,12 +33,22 @@ class BaseSpeedRuleElement extends RuleElementPF2e {
...
     override beforePrepareData(): void {
         if (this.ignored) return;
-        const speed = this.#createMovementType();
-        const synthetics = (this.actor.synthetics.movementTypes[this.selector] ??= []);
+        
+        const resolved = this.resolveInjectedProperties(this.selector);
+
+        let speedType: MovementType = "land"
+        if (!tupleHasValue(MOVEMENT_TYPES, resolved)) {
+            this.failValidation("Unrecognized or missing selector");
```
Comment (stwlam) [inline src/module/rules/rule-element/base-speed.ts:41]: You need to return here, or validation-failing data will get added to the synthetics object.

Comment (simonward) [reply src/module/rules/rule-element/base-speed.ts:41]: I figured we'd just add "land", same as original code. In any event, fixed.

File: src/module/rules/rule-element/base-speed.ts [hunk 2]
```diff
@@ -40,12 +33,23 @@ class BaseSpeedRuleElement extends RuleElementPF2e {
 
     override beforePrepareData(): void {
         if (this.ignored) return;
-        const speed = this.#createMovementType();
-        const synthetics = (this.actor.synthetics.movementTypes[this.selector] ??= []);
+
+        const resolved = this.resolveInjectedProperties(this.selector);
+
+        let speedType: MovementType;
```
Comment (stwlam) [inline src/module/rules/rule-element/base-speed.ts:39]: Don't use `let`

File: src/module/rules/rule-element/base-speed.ts [hunk 3]
```diff
@@ -40,12 +33,21 @@ class BaseSpeedRuleElement extends RuleElementPF2e {
...
+        const resolved = this.resolveInjectedProperties(this.selector);
+
+        if (!tupleHasValue(MOVEMENT_TYPES, resolved)) {
+            this.failValidation("Unrecognized or missing selector");
+            return;
+        }
+        const speedType: MovementType = resolved;
+
+        const speed = this.#createMovementType(speedType);
+        const synthetics = (this.actor.synthetics.movementTypes[speedType] ??= []);
```
Comment (stwlam) [inline src/module/rules/rule-element/base-speed.ts:46]:
```suggestion
        const speedType = this.resolveInjectedProperties(this.selector);
        if (!tupleHasValue(MOVEMENT_TYPES, speedType)) {
            return this.failValidation("Unrecognized or missing selector");
        }

        const speed = this.#createMovementType(speedType);
        const synthetics = (this.actor.synthetics.movementTypes[speedType] ??= []);
```

**Pull request merged**